R/with-cache.R

```diff
@@ -0,0 +1,522 @@
+#' Add caching with reactivity to an object
+#'
+#' @description
+#'
+#' `withCache()` adds caching to the following kinds of objects used in Shiny:
+#'
+#' * [reactive()]` expressions.
+#' * `render*` functions, like [renderText()], [renderTable()], and so on.
+#'
+#' It is often used in conjunction with [withEvent()].
+#'
+#'
+#' Ordinary [reactive()] expressions will cache their most recent value. This
```

Comment:
This description is a bit long I think; you probably want to keep it to 2-3 paragraphs.